cached_proc_macro/src/cached.rs

```diff
@@ -29,6 +29,8 @@ struct MacroArgs {
     #[darling(default)]
     sync_writes: bool,
```

Comment:
Change `sync_writes` to an Enum of possible values, with a `default` for darling.
Then pass nothing (no sync), defining with nothing (normal sync) or a string like `by_keys` (then by keys sync).
Then no need for a new flag.
I mean something like [this](https://github.com/async-graphql/async-graphql/blob/master/derive/src/args.rs#L44).

Reply:
But then `once` and `cached` have a different argument type for `sync_writes`, we could also add the enum for once, but it won't have any effect, as there is no key to cache by

Reply:
And this would be a breaking change

Reply:
Yes, it _can_ be a breaking change. (Sure, when it _can_, then it _is_ a breaking change!)
The `true` in `sync_writes = true` is optional. It can be `sync_writes` only without the boolean.
Even if users had that, it's not a hard to replace them.

And why cannot/shouldn't you add that to `once`?

Reply:
once doesn't have such key attribute, so sync by key doesn't make sense there, we could the attribute and just use the SyncWrite Mode that already exists.

Reply:
yeah can do that